ckanext/validation/plugin/__init__.py

```diff
@@ -344,3 +347,37 @@ def _get_underlying_file(wrapper):
         return wrapper.stream
     return wrapper.file
 
+
+def _should_remove_unsupported_resource_validation_reports(res_dict):
+    if not t.h.asbool(t.config.get('ckanext.validation.clean_validation_reports', False)):
+        return False
+    return (not res_dict.get('format', u'').lower() in settings.SUPPORTED_FORMATS
+            and (res_dict.get('url_type') == 'upload'
+                or res_dict.get('url_type') == '')
+            and (t.h.asbool(res_dict.get('validation_status', False))
+                or t.h.asbool(res_dict.get('extras', {}).get('validation_status', False))))
+
+
+def _remove_unsupported_resource_validation_reports(resource_id):
+    """
+    Callback to remove unsupported validation reports.
+    Controlled by config value: ckanext.validation.clean_validation_reports.
+    Double check the resource format. Only supported Validation formats should have validation reports.
+    If the resource format is not supported, we should delete the validation reports.
+    """
+    context = {"ignore_auth": True}
+    try:
+        res = p.toolkit.get_action('resource_show')(context, {"id": resource_id})
+    except t.ObjectNotFound:
+        log.error('Resource %s does not exist.', res['id'])
+        return
+
+    if _should_remove_unsupported_resource_validation_reports(res):
+        log.info('Unsupported resource format "%s". Deleting validation reports for resource %s',
+            res.get(u'format', u'').lower(), res['id'])
```

Comment:
We probably don't really need to lowercase the format.

Reply:
CKAN allows you to kind of put anything into the format field. And the validation extension has the supported format types of `csv` `xls` and `xlsx`. So a user could put the resource format as uppercase, but it would still be a valid csv file. So was just keeping with the check in `_should_remove_unsupported_resource_validation_reports`

But I can remove it from the log if we really don't care about it there?

Reply:
I think it's better to log the format exactly as the user entered it.

Reply:
Done